### PR TITLE
Grid search logging

### PR DESCRIPTION
**Pull request by SamVanhoutte**
Added ArcusGridSearch
Added LocalTrainer (without implementations)
Ready to merge

### CODE REVIEW

File: arcus/azureml/experimenting/aml_trainer.py [hunk 1]
```diff
@@ -28,69 +33,54 @@ def __init__(self, experiment_name: str, aml_workspace: Workspace):
...
-            # Save the model to the outputs directory for capture
-            model_file_name = 'outputs/model.pkl'
-            joblib.dump(value = fitted_model, filename = model_file_name)
-            self.__current_run.upload_file(name = model_file_name, path_or_stream = model_file_name)
-
-        if (finish_existing_run and self.__current_run is not None):
-            self.__current_run.complete(True)
-
-        return y_pred
+    def add_tuning_result(self, run_index: int, train_score: float, test_score: float, sample_count: int, durations:np.array, parameters: dict, estimator):
```
Comment (SriramAvatar) [inline arcus/azureml/experimenting/aml_trainer.py:65]: We have to add the Args information here?

File: arcus/azureml/experimenting/tuning.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
+from abc import ABCMeta, abstractmethod
+from sklearn.model_selection._search import BaseSearchCV
+from sklearn.model_selection._search import _check_param_grid
+from sklearn.model_selection._split import check_cv
+from sklearn.model_selection._validation import _fit_and_score
+from sklearn.metrics._scorer import _check_multimetric_scoring
+from sklearn.utils.validation import indexable, check_is_fitted, _check_fit_params
+from sklearn.base import is_classifier, clone, BaseEstimator
+is_classifier
```
Comment (SriramAvatar) [inline arcus/azureml/experimenting/tuning.py:9]: 'is_classifier', is this a typo?

File: .gitignore [hunk 3]
```diff
@@ -140,3 +140,4 @@ src/build/
 .azureml/
 samples/script_file.py
 samples/outputs/
+outputs/*
```
Comment (SriramAvatar) [inline .gitignore:143]: Is this outputs folder intended?

Comment (SamVanhoutte) [reply .gitignore:143]: yes, can appear in multiple directories (also test, etc)

File: arcus/azureml/experimenting/tuning.py (hunk 2 above)
Comment (SamVanhoutte) [reply arcus/azureml/experimenting/tuning.py:9]: good catch

**Pull request merged**